cadc-download-manager-server/src/main/java/ca/nrc/cadc/dlm/server/ServerUtil.java

```diff
@@ -153,28 +155,177 @@ public static String getCodebase(HttpServletRequest request) {
+                    for (String fileClass : fileClasses) {
+                        log.debug("fileClass: " + fileClass);
+                        sa = request.getParameterValues(fileClass);
+                        if (sa != null) {
+                            for (String curSa : sa) {
+                                String u = processURI(curSa);
+                                if (u != null) {
+                                    u = toAd(u);
+                                    log.debug("\turi: " + u);
+                                    uriList.add(new URI(u));
+                                }
+                            }
+                        }
+                    }
+                }
+
+                sa = request.getParameterValues("fileId");
+                if (sa != null) {
+                    log.debug("fileId param(s): " + sa.length);
+                    log.warn("deprecated param 'fileId' used by " + referer);
+                    for (String curSa : sa) {
+                        String u = processURI(curSa);
+                        if (u != null) {
+                            u = toAd(u);
+                            uriList.add(new URI(u));
+                        }
+                    }
+                }
+
+                if (!uriList.isEmpty()) {
+                    request.setAttribute("uriList", uriList);
+                }
+            } catch (URISyntaxException ure) {
+                log.error("error parsing URI from deprecated input parameter");
+            }
+        }
+
+        String params = (String) request.getAttribute("params");
+        if (params == null) {
+            Map<String,List<String>> paramMap = ServerUtil.getParameters(request);
+
+            // things to strip out
+            paramMap.remove("fileId");
+            List<String> fcs = paramMap.get("fileClass");
+            if (fcs != null) {
+                for (String fc : fcs) {
+                    paramMap.remove(fc);
+                }
+                paramMap.remove("fileClass");
+            }
+
+            List<String> frag = paramMap.get("fragment");
+            if (frag != null) {
+                for (String f : frag) {
+                    String[] parts = f.split("&");
+                    for (String p : parts) {
+                        String[] kv = p.split("=");
+                        if (kv.length == 2) {
+                            List<String> values = paramMap.get(kv[0]);
+                            if (values == null) {
+                                values = new ArrayList<String>();
+                                paramMap.put(kv[0], values);
+                            }
+                            values.add(kv[1]);
+                        }
+                    }
                 }
+                paramMap.remove("fragment");
+            }
+
+            if (!paramMap.isEmpty()) {
+                params = DownloadUtil.encodeParamMap(paramMap);
```

Comment:
IIRC, the idea of this encoded param map is the mechanism by  which the cutout, runid, etc get passed along -- it is not deprecated and should not be in here.

Reply:
moved parsing out of code area handling deprecated params.